app/src/main/java/com/vitaliidiadchenko/mykotkinapplication/FragmentMovieList.kt

```diff
@@ -47,17 +52,24 @@ class FragmentMovieList : Fragment() {
 
     private val movieListener = object : OnPosterCardClickListener {
         override fun onClick(movie: Movie) {
-            listener?.goToFragmentMoviesDetails()
+            listener?.goToFragmentMoviesDetails(movie)
+        }
+    }
+
+    private fun setMovies() {
+        scope.launch {
+            movies = context?.let { loadMovies(it)}
```

Comment:
Можно избавиться от ветвления в своем коде, воспользоваться встроенной функцией
```suggestion
            movies = loadMovies(requireContext())
```